src/hooks/useTimeout/index.js

```diff
@@ -0,0 +1,24 @@
+import { useCallback, useEffect, useState } from 'react';
+
+const useTimeout = () => {
+  const [timerIds, setTimerIds] = useState([]);
+
+  // clear the timeout.
+  useEffect(() => {
+    return () => timerIds.map((timerId) => clearTimeout(timerId));
```

Comment:
I know that it is very opinionable, also, that this way is very idiomatic for some JS people.

I am not asking for a change, only starting a conversation.

In my opinion, it is better to keep functions like `map` to _map_, it means to convert each item of a list into a different item in a new list without side effects.

`clearTimeout` is a effect, for that reason, I think that `for` will be clear here.

I know that it is not for JS ecosystem, but I think that [these Eric Lippert's reflections](https://ericlippert.com/2009/05/18/foreach-vs-foreach/) applyies here.

Reply:
I hadn't seen it that way, but it makes sense

Reply:
I guess it gives more expression to the code